usr/sbin/p11sak/p11sak.c

```diff
@@ -1655,7 +1659,27 @@ static CK_RV parse_list_key_args(char *argv[], int argc, p11sak_kt *kt,
             /* Get options */
         } else if (strcmp(argv[i], "--slot") == 0) {
             if (i + 1 < argc) {
-                *slot = (CK_ULONG) atol(argv[i + 1]);
+                str = argv[i + 1];
+                errno = 0;
+                *slot = strtol(str, &endptr, base);
```

Comment:
You could move the strtol handling plus error checking into a subroutine, since you have that same code 3 times now....
But if other reviewers are OK with having the same code 3 times, I am OK, too.

Reply:
i dont mind in that case.